src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1402,28 +1402,39 @@ InferenceProfiler::SummarizeSendRequestRate(
   summary.send_request_rate = num_sent_requests / window_duration_s;
 }
 
-int64_t
+cb::Error
 InferenceProfiler::DetermineStatsModelVersion(
     const cb::ModelIdentifier& model_identifier,
-    const std::map<cb::ModelIdentifier, cb::ModelStatistics>& stats)
+    const std::map<cb::ModelIdentifier, cb::ModelStatistics>& stats,
+    int64_t* status_model_version)
 {
-  // If model_version is an empty string then look in the end status to find
-  // the latest (highest valued version) and use that as the version.
-  int64_t status_model_version = -1;
+  // If model_version is an empty string then look in the stats to find the
+  // version with valid stats. If multiple versions have valid stats, use the
+  // highest numbered one and print a warning
+  *status_model_version = -1;
   bool multiple_found = false;
   if (model_identifier.second.empty()) {
     for (const auto& id : stats) {
-      // Model name should match
       if (model_identifier.first.compare(id.first.first) == 0) {
-        int64_t this_version = std::stoll(id.first.second);
-        if (status_model_version != -1) {
-          multiple_found = true;
+        if (id.second.inference_count_ > 0) {
```

Comment:
I think we can't rely on the zero inference count. For example, the user may have profiled this one in a previous run and started profiling the other version in the next run. If the inference count is the same before and after profiling that could be used as an indicator that this version was not profiled.

Reply:
updated it to factor in start and end stats

Reply:
Looks great!!